### DISCUSSION TITLE
Direct Memory Access (DMA)

### DISCUSSION THREAD

**Discussion by NNNILabs** [earlephilhower/arduino-pico#1314] [Q&A] [answered]
Hello, does Arduino Pico have DMA support? I checked the docs (https://arduino-pico.readthedocs.io/en/latest/), but I could only find references to DMA within certain audio libraries. 
Alternatively, would it be possible to use the C/C++ SDK headers and functions for DMA and have it compile and run successfully?
Regards, NNNI

**Comment by earlephilhower** [comment, marked as answer]
DMA is there but there is no wrapper library, so you need to use the SDK calls directly and handle interrupts/etc. yourself.  FWIW, though, the `AudioBufferManager` has covered a lot more use cases than I thought it would, and could be extended to a more general purpose DMA lib if someone had the time and inclination.  (hint hint 😆 )

>  Alternatively, would it be possible to use the C/C++ SDK headers and functions for DMA and have it compile and run successfully?

Yes, they're all included in the path and static libs.  That's how the AudioBufferManager works.

**Comment by NNNILabs** [comment]
Thanks a lot for the answer!
